enterprise/server/ociregistry/ociregistry.go

```diff
@@ -245,32 +301,62 @@ func (r *registry) handleBlobRequest(ctx context.Context, w http.ResponseWriter,
 		}()
 	}
 
-	rc, err := layer.Compressed()
-	if err != nil {
-		http.Error(w, fmt.Sprintf("could not create blob reader: %s", err), http.StatusInternalServerError)
-		return
-	}
-	defer rc.Close()
-
-	if offset > 0 {
-		io.CopyN(io.Discard, rc, offset)
+	var casWriter io.Writer
+	fetchingFullLayer := offset == 0 && (limit == 0 || limit >= length)
+	if fetchingFullLayer {
+		casWriter = w
+	} else {
+		casWriter, err = newStartEndWriter(w, offset, offset+limit)
```

Comment:
There are simpler alternatives you could employ here:
 1. The ByteStream API supports offets and limits, but we currently don't expose that as part of the `cachetools.GetBlob` function. You could add another function for that, something like `cachetools.GetPartialBlob`. 
 2. You can use go pipes to create a reader that's connected to a writer (https://pkg.go.dev/io#Pipe). Then you can copy the first N bytes you are not interested into a discarding wrtiter. Then copy the the remaining bytes you are interested in into the http writer.

I think (1) is the better way to go since it reduces the amount of bytes that need to be transferred between from the cache.

Reply:
Thanks for those suggestions! I implemented `GetPartialBlob`.
I could also make sure to only request the necessary bytes from the remote registry.